### PR TITLE
#110 Add coroutine dispatcher qualifier

### LINKED ISSUE #110: Create a class for handling code and test dispatchers

### PR DESCRIPTION
**Pull request by MarekMacko**

### CODE REVIEW

File: app/src/main/kotlin/com/intive/picover/common/coroutines/dispatcher/Dispatcher.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.intive.picover.common.coroutines.dispatcher
+
+import javax.inject.Qualifier
+
+annotation class Dispatcher {
+
+	@Retention(AnnotationRetention.BINARY)
```
Comment (michal-kucznerowicz) [inline app/src/main/kotlin/com/intive/picover/common/coroutines/dispatcher/Dispatcher.kt]: Why do we need to retain this annotation in binary output?

File: app/src/main/kotlin/com/intive/picover/common/coroutines/dispatcher/Dispatcher.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+	@Qualifier
+	annotation class IO
```
Comment (michal-kucznerowicz) [inline app/src/main/kotlin/com/intive/picover/common/coroutines/dispatcher/Dispatcher.kt:10]: You can also specify annotation target e.g. `@Target(AnnotationTarget.PROPERTY, AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.FUNCTION)`

**Review by michal-kucznerowicz — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/com/intive/picover/common/coroutines/dispatcher/Dispatcher.kt (hunk 1 above)
Comment (MarekMacko) [reply app/src/main/kotlin/com/intive/picover/common/coroutines/dispatcher/Dispatcher.kt]: That's the review I was looking for 😄 Good point, I'll change it to `SOURCE`

File: app/src/main/kotlin/com/intive/picover/common/coroutines/dispatcher/Dispatcher.kt (hunk 2 above)
Comment (MarekMacko) [reply app/src/main/kotlin/com/intive/picover/common/coroutines/dispatcher/Dispatcher.kt:10]: I'll set it only to `@Target(AnnotationTarget.FUNCTION, AnnotationTarget.VALUE_PARAMETER)`, `PROPERTY` is not needed at this moment

**Review by michal-kucznerowicz — APPROVED** [review]

**Review by marcinWisnia — APPROVED** [review]

**Pull request merged**